### ISSUE TITLE
`Embedding::forward` produces no gradient for weight on GPU

### ISSUE THREAD

**Issue by ngutten** [dollspace-gay/ferrotorch#11]
When `Embedding::forward` runs on a GPU tensor, the backward pass does not produce a gradient for `embedding.weight`. The forward output is correct, and downstream gradients flow normally — only the embedding weight itself is affected.

```rust
let emb = Embedding::new(1000, 64, None)?;
emb.to_device(Device::Cuda(0))?;

let ids = Tensor::<f32>::from_storage(
    TensorStorage::cpu(vec![5.0, 10.0, 15.0]),
    vec![3], false,
)?.to(Device::Cuda(0))?;

let out = emb.forward(&ids)?;
let loss = out.sum_all()?;
loss.backward()?;
emb.weight.tensor().grad() // → None
```

On CPU, the gradient is sparse (nonzero only at the looked-up indices), which is correct. On GPU, it's absent entirely. Likely the Embedding backward doesn't have a GPU code path.
